src/datagrid/plugins/types.ts

```diff
@@ -1,8 +1,10 @@
-import type { Cell } from './cell-selection';
+import type { CellIndexes } from './cell-selection';
+import type { RefObject } from 'react';
+import type { PluginHandle } from './createPlugin';
 
 export type Selection = {
-    start: Cell;
-    end: Cell;
+    start: CellIndexes;
+    end: CellIndexes;
```

Comment:
end and start include this includes IDs but isn't reflected here.
always pass an array of columnIds and rowIds in events. ids of rows/columns, start and end inclusive.